src/main/java/com/sandflow/smpte/regxml/FragmentBuilder.java

```diff
@@ -991,7 +991,13 @@ void applyRule5_2(Element element, MXFInputStream value, EnumerationTypeDefiniti
 
             byte[] val = new byte[len];
 
-            int br = value.read(val);
```

Comment:
Why doesn't `readFully()` work here as well?

Reply:
It looks like the array is still used even if the length is less than expected (https://github.com/sandflow/regxmllib/pull/138/files#diff-e348406d2dd6006f6c9735f2f1538f1aR1024) so you'd have to catch (and ignore) the `EOFException`.

You also use the number of bytes read, which you don't get from `readFully()`, in the `if` condition below (although I think you could replace that with `val.length`).

Reply:
Now that I think about it, are you sure you want to pass a partially read array into `new BigInteger()`? If the array has been partially read then it will be right padded with zeros so it will be a different value than if the same bytes had been read into to smaller array.
```java
byte[] full = {1}; // one byte read into a one byte array
System.out.println(new BigInteger(full)); // 1
byte[] partial = {1, 0}; // one byte read into a two byte array
System.out.println(new BigInteger(partial)); // 256
```

Reply:
Understood. I like your proposals to: (a) replace the single `read` by the loop and (b) pass the actual read number of bytes to `BigInteger()`.

Reply:
I've added a commit to pass in the actual read number of bytes to `BigInteger`. The `BigInteger` constructor that takes a byte range was added in Java 9 so I've used `Arrays.copyOf` to truncate the array to the correct size in order to maintain Java 8 compatibility.